source/extensions/common/wasm/plugin.cc

```diff
@@ -20,6 +14,35 @@ WasmConfig::WasmConfig(const envoy::extensions::wasm::v3::PluginConfig& config)
     // TODO(rapilado): Set the SanitizationConfig fields once sanitization is implemented.
     allowed_capabilities_[capability.first] = proxy_wasm::SanitizationConfig();
   }
+
+  if (config_.vm_config().has_environment_variables()) {
```

Comment:
Actually, perhaps we should be rejecting configurations that use NullVM and set environment variables?